<a href="https://colab.research.google.com/github/SohamIITK/Physics-Extras/blob/main/PHY462_GM_Counter.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np

Calculations

### Dead Time

In [ ]:
m1 = 27.215
m2 = 0.51
m12 = 27.315
mb = 0.125

X = m1*m2 - mb*m12
Y = m1*m2*(m12 + mb) - mb*m12*(m1+m2)
Z = Y*(m1 + m2 - m12 - mb)/X**2

print("X= ", X)
print("Y = ", Y)
print("Z = ", Z)

X=  10.465275
Y =  286.194049125
Z =  0.7447391234686388


In [ ]:
dead_time = X*(1-np.sqrt(1-Z))/Y
print("Dead Time is", dead_time)

Dead Time is 0.018092156820391902


## Poisson Stats (Background Counts)

In [ ]:
y = np.array([0, 1, 2, 3, 4, 5, 6, 7 , 8, 9, 10, 11, 12])
fy = np.array([4, 4, 35, 39, 56, 27, 23, 12, 1, 4, 0, 2, 1])


print(np.sum(fy))

208


In [ ]:
y_bar = np.sum(y*fy)/np.sum(fy)
print("Mean is ", y_bar)

for x in range(0, len(y)):
    probs = (y_bar**y[x] * np.exp(-y_bar))/math.factorial(x)
    print("Prob. for y = ", y[x], "is", probs)
# probs = (y_bar**y * np.exp(-y_bar))/math.factorial(x)
# print(probs)

std_dev_bg = np.sqrt(y_bar)  # theoretical std. dev (Poisson)
print("Std. Dev for BG counts = ", std_dev_bg)

Mean is  4.086538461538462
Prob. for y =  0 is 0.016797277462004382
Prob. for y =  1 is 0.06864272039761406
Prob. for y =  2 is 0.14025555850474028
Prob. for y =  3 is 0.19105324475805965
Prob. for y =  4 is 0.19518660822638312
Prob. for y =  5 is 0.15952751633887083
Prob. for y =  6 is 0.10865255519875015
Prob. for y =  7 is 0.06343040653773188
Prob. for y =  8 is 0.032401349493432756
Prob. for y =  9 is 0.014712151212295858
Prob. for y =  10 is 0.006012177178101673
Prob. for y =  11 is 0.002233544843263296
Prob. for y =  12 is 0.0007606222422971963
Std. Dev for BG counts =  2.0215188501565997


In [ ]:
y_poiss_var = (np.sum((y - y_bar)**2)/(len(y) - 1))
print("Variance of background counts is ", y_poiss_var) # experimental var (Poisson)

y_poiss_std_dev = np.sqrt(np.sum((y - y_bar)**2)/(len(y) - 1))
print("Std. Dev of background counts is ", y_poiss_std_dev) # experimental std dev (Poisson)

Variance of background counts is  19.13311298076923
Std. Dev of background counts is  4.374141399265601


## Gaussian Dist


In [ ]:
y_gauss = np.array([1037,964,1026,1014,1009,1016,1049,1026,1044,985,1005,956,1022,1012,1033,1022,1016,985,1019,1026,1002,971,1000,1027,1017,1000,1000,1039,1031,991,999,1051,1044,979,1024,1013,1013,1033,1034,1021,1025,980,1026,975,982,1014,1025,993,1006,1008,997,1032,1019,1026,1020,1042,1033,1045,989,1030,1021,1029,1016,1028,991,1026,989,1004,1012,996,1002,1068,1007,1034,1004,1014,1004,980,1039,1019,1009,1043,1038,999,1006,1027,988,997,1005,1026,1001,1045,1041,1025,1043,1013,1024,976,984,1003])

In [ ]:
y_gauss_mean = np.sum(y_gauss)/len(y_gauss)
print("Mean of Cs-137 counts is ", y_gauss_mean) # experimental mean

y_gauss_var = (np.sum((y_gauss - y_gauss_mean)**2)/(len(y_gauss) - 1))
print("Var of Cs-137 counts is ", y_gauss_var) # experimental variance (Gaussian)

y_gauss_std_dev = np.sqrt(np.sum((y_gauss - y_gauss_mean)**2)/(len(y_gauss) - 1))
print("Std. Dev of Cs-137 counts is ", y_gauss_std_dev) # experimental std. dev (Gaussian)

Mean of Cs-137 counts is  1013.98
Var of Cs-137 counts is  451.7369696969696
Std. Dev of Cs-137 counts is  21.254104772889626


Variance	$s^2$ =	451.73697

Standard Deviation	s =	21.254105

Count	n =	100

Mean	$y_{mean}$=	1013.98

Sum of Squares	SS =	44721.96


## Absorption Coeffs

### Al


In [ ]:
Na = 6.023*10**26

ne_Al = 13 * (2.7 * 10**3) * Na / 27
print("For Al, no. of electrons per unit volume n_e = ", ne_Al)

mu_Al = (2.466+0.86)/2
cross_section_Al = mu_Al/ne_Al
print("Cross section for Al = ", cross_section_Al * 10**28, "barn")

For Al, no. of electrons per unit volume n_e =  7.8299e+29
Cross section for Al =  0.021239096284754597 barn


### Cu

In [ ]:
ne_Cu = 29 * (8.92 * 10**3) * Na / 63.5
print("For Cu, no. of electrons per unit volume n_e = ", ne_Cu)

mu_Cu = (1.703+4.294)/2
cross_section_Cu = mu_Cu/ne_Cu
print("Cross section for Cu = ", cross_section_Cu * 10**28, "barn")

For Cu, no. of electrons per unit volume n_e =  2.4535899842519685e+30
Cross section for Cu =  0.012220868275651033 barn
